204 Updated SUGCON sites to latest version of Headless Template

robearlam opened the pull request:
This commit pulls in all of the changes from XM Cloud Foundation Head. As we took our cut a long time ago, pretty much every component & asset has changed since then. I've tried to unpick any customisations we've made to bring us a closely in line with the base repo as possible.

## Description / Motivation
* Pulls in all code & config changes from the XM Cloud Foundation Head repo
* Installs missing SXA Modules (Editing Theme and FEaaS)
* Changes Existing AppRoute template to Page to match base definition
* Updates init and README to now include CDP params that are required

## Types of changes
- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have read the Contributing guide.
- [x] My code/comments/docs fully adhere to the Code of Conduct.
- [x] My change is a code change.
- [ ] My change is a documentation change and there are NO other updates required.

Closes #204

sebastiantecsi commented:
Hi @robearlam ,

I had a look at the changes that you made, and there are couple of things:

1. https://sugconanz.xmcloudcm.localhost/ does not load at all,
I see this logs messages in docker:
```
2023-05-08 14:26:35 npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules\fsevents):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-ia32-msvc@13.2.4 (node_modules\@next\swc-win32-ia32-msvc):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-ia32-msvc@13.2.4: wanted {"os":"win32","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-arm64-msvc@13.2.4 (node_modules\@next\swc-win32-arm64-msvc):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-arm64-msvc@13.2.4: wanted {"os":"win32","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-x64-musl@13.2.4 (node_modules\@next\swc-linux-x64-musl):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-x64-musl@13.2.4: wanted {"os":"linux","arch":"x64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-x64-gnu@13.2.4 (node_modules\@next\swc-linux-x64-gnu):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-x64-gnu@13.2.4: wanted {"os":"linux","arch":"x64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm64-musl@13.2.4 (node_modules\@next\swc-linux-arm64-musl):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm64-musl@13.2.4: wanted {"os":"linux","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm64-gnu@13.2.4 (node_modules\@next\swc-linux-arm64-gnu):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm64-gnu@13.2.4: wanted {"os":"linux","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-darwin-x64@13.2.4 (node_modules\@next\swc-darwin-x64):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-darwin-x64@13.2.4: wanted {"os":"darwin","arch":"x64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm-gnueabihf@13.2.4 (node_modules\@next\swc-linux-arm-gnueabihf):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm-gnueabihf@13.2.4: wanted {"os":"linux","arch":"arm"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-freebsd-x64@13.2.4 (node_modules\@next\swc-freebsd-x64):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-freebsd-x64@13.2.4: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-darwin-arm64@13.2.4 (node_modules\@next\swc-darwin-arm64):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-darwin-arm64@13.2.4: wanted {"os":"darwin","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-android-arm64@13.2.4 (node_modules\@next\swc-android-arm64):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-android-arm64@13.2.4: wanted {"os":"android","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-android-arm-eabi@13.2.4 (node_modules\@next\swc-android-arm-eabi):
2023-05-08 14:30:29 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-android-arm-eabi@13.2.4: wanted {"os":"android","arch":"arm"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:30:29 
2023-05-08 14:30:33 npm WARN lifecycle npm is using C:\node\node.exe but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
2023-05-08 14:30:29 added 1096 packages from 680 contributors and audited 1117 packages in 233.765s
2023-05-08 14:30:31 
2023-05-08 14:30:31 159 packages are looking for funding
2023-05-08 14:30:31   run `npm fund` for details
2023-05-08 14:30:31 
2023-05-08 14:30:31 found 20 vulnerabilities (11 moderate, 6 high, 3 critical)
2023-05-08 14:30:31   run `npm audit fix` to fix them, or `npm audit` for details
2023-05-08 14:30:33 
2023-05-08 14:30:33 > sugconanz@21.1.1 start:connected C:\app
2023-05-08 14:30:33 > npm-run-all --serial bootstrap --parallel next:dev start:watch-components
2023-05-08 14:30:33 
2023-05-08 14:30:36 
2023-05-08 14:30:36 > sugconanz@21.1.1 bootstrap C:\app
2023-05-08 14:30:36 > ts-node --project tsconfig.scripts.json scripts/bootstrap.ts
2023-05-08 14:30:36 
2023-05-08 14:30:45 Registering config plugin computed
2023-05-08 14:30:45 Registering config plugin fallback
2023-05-08 14:30:45 Registering config plugin multisite
2023-05-08 14:30:45 Registering config plugin package-json
2023-05-08 14:30:45 Registering config plugin scjssconfig
2023-05-08 14:30:45 Writing config plugins to C:\app\scripts\temp\config-plugins.ts
2023-05-08 14:30:45 Registering sitemap-fetcher plugin graphql-sitemap-service
2023-05-08 14:30:45 Writing sitemap-fetcher plugins to C:\app\src\temp\sitemap-fetcher-plugins.ts
2023-05-08 14:30:45 Registering middleware plugin multisite
2023-05-08 14:30:45 Registering middleware plugin personalize
2023-05-08 14:30:36 npm WARN lifecycle npm is using C:\node\node.exe but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
2023-05-08 14:30:45 Registering middleware plugin redirects
2023-05-08 14:30:45 Writing middleware plugins to C:\app\src\temp\middleware-plugins.ts
2023-05-08 14:30:45 Registering page-props-factory plugin component-props
2023-05-08 14:30:45 Registering page-props-factory plugin error-pages
2023-05-08 14:30:45 Registering page-props-factory plugin normal-mode
2023-05-08 14:30:45 Registering page-props-factory plugin personalize
2023-05-08 14:30:45 Registering page-props-factory plugin preview-mode
2023-05-08 14:30:45 Registering page-props-factory plugin site
2023-05-08 14:30:45 Writing page-props-factory plugins to C:\app\src\temp\page-props-factory-plugins.ts
2023-05-08 14:30:45 Registering next-config plugin graphql
2023-05-08 14:30:45 Registering next-config plugin robots
2023-05-08 14:30:45 Registering next-config plugin sass
2023-05-08 14:30:45 Registering next-config plugin sitemap
2023-05-08 14:30:45 Writing next-config plugins to C:\app\src\temp\next-config-plugins.js
2023-05-08 14:30:45 Registering extract-path plugin multisite
2023-05-08 14:30:45 Registering extract-path plugin personalize
2023-05-08 14:30:45 Writing extract-path plugins to C:\app\src\temp\extract-path-plugins.ts
2023-05-08 14:30:45 Registering site-resolver plugin default
2023-05-08 14:30:45 Registering site-resolver plugin multisite
2023-05-08 14:30:45 Writing site-resolver plugins to C:\app\src\temp\site-resolver-plugins.ts
2023-05-08 14:30:52 Registering JSS component CdpPageView
2023-05-08 14:30:52 Registering JSS component ColumnSplitter
2023-05-08 14:30:52 Registering JSS component Container
2023-05-08 14:30:52 Registering JSS component ContentBlock
2023-05-08 14:30:52 Registering JSS component FEaaSWrapper
2023-05-08 14:30:52 Registering JSS component Image
2023-05-08 14:30:52 Registering JSS component LinkList
2023-05-08 14:30:52 Registering JSS component Navigation
2023-05-08 14:30:52 Registering JSS component PageContent
2023-05-08 14:30:52 Registering JSS component PartialDesignDynamicPlaceholder
2023-05-08 14:30:52 Registering JSS component Promo
2023-05-08 14:30:52 Registering JSS component RichText
2023-05-08 14:30:52 Registering JSS component RowSplitter
2023-05-08 14:30:52 Registering JSS component Title
2023-05-08 14:30:52 Registering JSS component Agenda
2023-05-08 14:30:52 Registering JSS component CTA
2023-05-08 14:30:52 Registering JSS component FixedWidthContainer
2023-05-08 14:30:52 Registering JSS component Hero
2023-05-08 14:30:52 Registering JSS component OrganiserList
2023-05-08 14:30:52 Registering JSS component Session
2023-05-08 14:30:52 Registering JSS component Speaker
2023-05-08 14:30:52 Registering JSS component SponsorList
2023-05-08 14:30:52 Writing component factory to C:\app\src\temp\componentFactory.ts
2023-05-08 14:30:52 Writing runtime config to C:\app\src\temp\config.js
2023-05-08 14:30:53 
2023-05-08 14:30:53 > sugconanz@21.1.1 start:watch-components C:\app
2023-05-08 14:30:53 > ts-node --project tsconfig.scripts.json scripts/generate-component-factory.ts --watch
2023-05-08 14:30:53 
2023-05-08 14:30:53 
2023-05-08 14:30:53 > sugconanz@21.1.1 next:dev C:\app
2023-05-08 14:30:52 Skipping site information fetch (missing GraphQL endpoint or API key).
2023-05-08 14:30:53 npm WARN lifecycle npm is using C:\node\node.exe but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
2023-05-08 14:30:53 npm WARN lifecycle npm is using C:\node\node.exe but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
2023-05-08 14:30:53 Debugger listening on ws://127.0.0.1:9229/17b2198c-0394-4dd7-a759-2ef4e81e9dca
2023-05-08 14:30:53 For help, see: https://nodejs.org/en/docs/inspector
2023-05-08 14:30:55 warn  - the --inspect option was detected, the Next.js server should be inspected at port 9230.
2023-05-08 14:30:55 Debugger listening on ws://127.0.0.1:9230/97c1a1b1-612a-4c7d-a43c-4fd462f051f7
2023-05-08 14:30:55 For help, see: https://nodejs.org/en/docs/inspector
2023-05-08 14:31:04 Starting inspector on 127.0.0.1:9229 failed: address already in use
2023-05-08 14:31:04 Starting inspector on 127.0.0.1:9229 failed: address already in use
2023-05-08 14:31:07 Starting inspector on 127.0.0.1:9229 failed: address already in use
2023-05-08 14:31:07 Starting inspector on 127.0.0.1:9229 failed: address already in use
2023-05-08 14:31:11 warn  - Attempted to load @next/swc-win32-x64-gnu, but it was not installed
2023-05-08 14:30:53 > cross-env NODE_OPTIONS='--inspect' next dev
2023-05-08 14:30:53 
2023-05-08 14:30:59 ready - started server on 0.0.0.0:3000, url: http://localhost:3000
2023-05-08 14:31:01 Watching for changes to component factory sources in src/components...
2023-05-08 14:31:03 Attention: Next.js now collects completely anonymous telemetry regarding usage.
2023-05-08 14:31:03 This information is used to shape Next.js' roadmap and prioritize features.
2023-05-08 14:31:03 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
2023-05-08 14:31:03 https://nextjs.org/telemetry
2023-05-08 14:31:03 
2023-05-08 14:31:06 info  - Downloading WASM swc package...
2023-05-08 14:31:11 info  - Using wasm build of next-swc
2023-05-08 14:31:12 info  - automatically enabled Fast Refresh for 1 custom loader
2023-05-08 14:31:11 warn  - Attempted to load @next/swc-win32-x64-msvc, but an error occurred: The specified module could not be found.
2023-05-08 14:31:11 \\?\C:\app\node_modules\@next\swc-win32-x64-msvc\next-swc.win32-x64-msvc.node

```
2. https://sugconeu.xmcloudcm.localhost/
I see this logs messages in docker:
```
2023-05-08 14:26:35 npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
2023-05-08 14:29:49 npm ERR! cb() never called!
2023-05-08 14:29:49 
2023-05-08 14:29:49 npm ERR! This is an error with npm itself. Please report this error at:
2023-05-08 14:29:49 npm ERR!     <https://npm.community>
2023-05-08 14:29:49 
2023-05-08 14:29:49 npm ERR! A complete log of this run can be found in:
2023-05-08 14:29:49 npm ERR!     C:\Users\ContainerUser\AppData\Roaming\npm-cache\_logs\2023-05-08T11_29_49_541Z-debug.log
2023-05-08 14:46:24 npm WARN read-shrinkwrap This version of npm is compatible with lockfileVersion@1, but package-lock.json was generated for lockfileVersion@2. I'll try to do my best with it!
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: fsevents@2.3.2 (node_modules\fsevents):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for fsevents@2.3.2: wanted {"os":"darwin","arch":"any"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-ia32-msvc@13.2.4 (node_modules\@next\swc-win32-ia32-msvc):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-ia32-msvc@13.2.4: wanted {"os":"win32","arch":"ia32"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-win32-arm64-msvc@13.2.4 (node_modules\@next\swc-win32-arm64-msvc):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-win32-arm64-msvc@13.2.4: wanted {"os":"win32","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-x64-musl@13.2.4 (node_modules\@next\swc-linux-x64-musl):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-x64-musl@13.2.4: wanted {"os":"linux","arch":"x64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-x64-gnu@13.2.4 (node_modules\@next\swc-linux-x64-gnu):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-x64-gnu@13.2.4: wanted {"os":"linux","arch":"x64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm64-musl@13.2.4 (node_modules\@next\swc-linux-arm64-musl):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm64-musl@13.2.4: wanted {"os":"linux","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm64-gnu@13.2.4 (node_modules\@next\swc-linux-arm64-gnu):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm64-gnu@13.2.4: wanted {"os":"linux","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-linux-arm-gnueabihf@13.2.4 (node_modules\@next\swc-linux-arm-gnueabihf):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-linux-arm-gnueabihf@13.2.4: wanted {"os":"linux","arch":"arm"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-freebsd-x64@13.2.4 (node_modules\@next\swc-freebsd-x64):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-freebsd-x64@13.2.4: wanted {"os":"freebsd","arch":"x64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-darwin-arm64@13.2.4 (node_modules\@next\swc-darwin-arm64):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-darwin-arm64@13.2.4: wanted {"os":"darwin","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-darwin-x64@13.2.4 (node_modules\@next\swc-darwin-x64):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-darwin-x64@13.2.4: wanted {"os":"darwin","arch":"x64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-android-arm64@13.2.4 (node_modules\@next\swc-android-arm64):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-android-arm64@13.2.4: wanted {"os":"android","arch":"arm64"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 npm WARN optional SKIPPING OPTIONAL DEPENDENCY: @next/swc-android-arm-eabi@13.2.4 (node_modules\@next\swc-android-arm-eabi):
2023-05-08 14:48:36 npm WARN notsup SKIPPING OPTIONAL DEPENDENCY: Unsupported platform for @next/swc-android-arm-eabi@13.2.4: wanted {"os":"android","arch":"arm"} (current: {"os":"win32","arch":"x64"})
2023-05-08 14:48:36 
2023-05-08 14:48:36 added 1096 packages from 680 contributors and audited 1117 packages in 132.075s
2023-05-08 14:48:37 
2023-05-08 14:48:37 159 packages are looking for funding
2023-05-08 14:48:37   run `npm fund` for details
2023-05-08 14:48:37 
2023-05-08 14:48:37 found 20 vulnerabilities (11 moderate, 6 high, 3 critical)
2023-05-08 14:48:37   run `npm audit fix` to fix them, or `npm audit` for details
2023-05-08 14:48:39 npm WARN lifecycle npm is using C:\node\node.exe but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
2023-05-08 14:48:39 
2023-05-08 14:48:39 > sugconeu@21.1.1 start:connected C:\app
2023-05-08 14:48:39 > npm-run-all --serial bootstrap --parallel next:dev start:watch-components
2023-05-08 14:48:39 
2023-05-08 14:48:41 npm WARN lifecycle npm is using C:\node\node.exe but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
2023-05-08 14:48:41 
2023-05-08 14:48:41 > sugconeu@21.1.1 bootstrap C:\app
2023-05-08 14:48:41 > ts-node --project tsconfig.scripts.json scripts/bootstrap.ts
2023-05-08 14:48:41 
2023-05-08 14:48:47 Registering config plugin computed
2023-05-08 14:48:47 Registering config plugin fallback
2023-05-08 14:48:47 Registering config plugin multisite
2023-05-08 14:48:47 Registering config plugin package-json
2023-05-08 14:48:47 Registering config plugin scjssconfig
2023-05-08 14:48:47 Writing config plugins to C:\app\scripts\temp\config-plugins.ts
2023-05-08 14:48:47 Registering sitemap-fetcher plugin graphql-sitemap-service
2023-05-08 14:48:47 Writing sitemap-fetcher plugins to C:\app\src\temp\sitemap-fetcher-plugins.ts
2023-05-08 14:48:47 Registering middleware plugin multisite
2023-05-08 14:48:47 Registering middleware plugin personalize
2023-05-08 14:48:47 Registering middleware plugin redirects
2023-05-08 14:48:47 Writing middleware plugins to C:\app\src\temp\middleware-plugins.ts
2023-05-08 14:48:47 Registering page-props-factory plugin component-props
2023-05-08 14:48:47 Registering page-props-factory plugin error-pages
2023-05-08 14:48:47 Registering page-props-factory plugin normal-mode
2023-05-08 14:48:47 Registering page-props-factory plugin personalize
2023-05-08 14:48:47 Registering page-props-factory plugin preview-mode
2023-05-08 14:48:47 Registering page-props-factory plugin site
2023-05-08 14:48:47 Writing page-props-factory plugins to C:\app\src\temp\page-props-factory-plugins.ts
2023-05-08 14:48:47 Registering next-config plugin graphql
2023-05-08 14:48:47 Registering next-config plugin robots
2023-05-08 14:48:47 Registering next-config plugin sass
2023-05-08 14:48:47 Registering next-config plugin sitemap
2023-05-08 14:48:47 Writing next-config plugins to C:\app\src\temp\next-config-plugins.js
2023-05-08 14:48:47 Registering extract-path plugin multisite
2023-05-08 14:48:47 Registering extract-path plugin personalize
2023-05-08 14:48:47 Writing extract-path plugins to C:\app\src\temp\extract-path-plugins.ts
2023-05-08 14:48:47 Registering site-resolver plugin default
2023-05-08 14:48:47 Registering site-resolver plugin multisite
2023-05-08 14:48:47 Writing site-resolver plugins to C:\app\src\temp\site-resolver-plugins.ts
2023-05-08 14:48:53 Registering JSS component CdpPageView
2023-05-08 14:48:53 Registering JSS component ColumnSplitter
2023-05-08 14:48:53 Registering JSS component Container
2023-05-08 14:48:53 Registering JSS component ContentBlock
2023-05-08 14:48:53 Registering JSS component FEaaSWrapper
2023-05-08 14:48:53 Registering JSS component Image
2023-05-08 14:48:53 Registering JSS component LinkList
2023-05-08 14:48:53 Registering JSS component Navigation
2023-05-08 14:48:53 Registering JSS component PageContent
2023-05-08 14:48:53 Registering JSS component PartialDesignDynamicPlaceholder
2023-05-08 14:48:53 Registering JSS component Promo
2023-05-08 14:48:53 Registering JSS component RichText
2023-05-08 14:48:53 Registering JSS component RowSplitter
2023-05-08 14:48:53 Registering JSS component Title
2023-05-08 14:48:53 Registering JSS component Agenda
2023-05-08 14:48:53 Registering JSS component AnnouncementBar
2023-05-08 14:48:53 Registering JSS component CTA
2023-05-08 14:48:53 Registering JSS component FixedWidthContainer
2023-05-08 14:48:53 Registering JSS component Hero
2023-05-08 14:48:53 Registering JSS component OrganiserList
2023-05-08 14:48:53 Registering JSS component Session
2023-05-08 14:48:53 Registering JSS component Speaker
2023-05-08 14:48:53 Registering JSS component SponsorList
2023-05-08 14:48:53 Writing component factory to C:\app\src\temp\componentFactory.ts
2023-05-08 14:48:53 Writing runtime config to C:\app\src\temp\config.js
2023-05-08 14:48:53 Skipping site information fetch (missing GraphQL endpoint or API key).
2023-05-08 14:48:54 
2023-05-08 14:48:54 > sugconeu@21.1.1 next:dev C:\app
2023-05-08 14:48:54 > cross-env NODE_OPTIONS='--inspect' next dev
2023-05-08 14:48:54 
2023-05-08 14:48:54 
2023-05-08 14:48:54 > sugconeu@21.1.1 start:watch-components C:\app
2023-05-08 14:48:54 npm WARN lifecycle npm is using C:\node\node.exe but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
2023-05-08 14:48:54 npm WARN lifecycle npm is using C:\node\node.exe but there is no node binary in the current PATH. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.
2023-05-08 14:48:54 > ts-node --project tsconfig.scripts.json scripts/generate-component-factory.ts --watch
2023-05-08 14:48:54 
2023-05-08 14:48:55 Debugger listening on ws://127.0.0.1:9229/8ab04ea4-652d-4b5a-bca5-376d36393ba3
2023-05-08 14:48:55 For help, see: https://nodejs.org/en/docs/inspector
2023-05-08 14:48:56 warn  - the --inspect option was detected, the Next.js server should be inspected at port 9230.
2023-05-08 14:48:57 Debugger listening on ws://127.0.0.1:9230/24e90c97-34bc-4221-9951-239587f9c825
2023-05-08 14:48:57 For help, see: https://nodejs.org/en/docs/inspector
2023-05-08 14:49:02 ready - started server on 0.0.0.0:3000, url: http://localhost:3000
2023-05-08 14:49:03 Watching for changes to component factory sources in src/components...
2023-05-08 14:49:06 Attention: Next.js now collects completely anonymous telemetry regarding usage.
2023-05-08 14:49:06 This information is used to shape Next.js' roadmap and prioritize features.
2023-05-08 14:49:06 You can learn more, including how to opt-out if you'd not like to participate in this anonymous program, by visiting the following URL:
2023-05-08 14:49:06 https://nextjs.org/telemetry
2023-05-08 14:49:06 
2023-05-08 14:49:07 Starting inspector on 127.0.0.1:9229 failed: address already in use
2023-05-08 14:49:07 Starting inspector on 127.0.0.1:9229 failed: address already in use
2023-05-08 14:49:09 info  - Downloading WASM swc package...
2023-05-08 14:49:10 Starting inspector on 127.0.0.1:9229 failed: address already in use
2023-05-08 14:49:10 Starting inspector on 127.0.0.1:9229 failed: address already in use
2023-05-08 14:49:13 info  - Using wasm build of next-swc
2023-05-08 14:49:13 warn  - Attempted to load @next/swc-win32-x64-gnu, but it was not installed
2023-05-08 14:49:13 warn  - Attempted to load @next/swc-win32-x64-msvc, but an error occurred: The specified module could not be found.
2023-05-08 14:49:13 \\?\C:\app\node_modules\@next\swc-win32-x64-msvc\next-swc.win32-x64-msvc.node
2023-05-08 14:49:14 info  - automatically enabled Fast Refresh for 1 custom loader

```

3. src/Project/Sugcon/SugconAnzSxa/package-lock.json  is not the latest in the repo/branch
4. src/Project/Sugcon/SugconEUSxa/package-lock.json  is not the latest in the repo/branch

robearlam commented:
I don't see any of those issues locally @sebastiantecsi.

It feels like you might need to do a clean of the repo on your local filesystem after changing to the new branch. According to this [SO post](https://stackoverflow.com/questions/69816589/next-failed-to-load-swc-binary), the `Attempted to load @next/swc-win32-x64-msvc` error (1, & 2 in your list above) is related to mismatched package versions in the `node_modules` folder, which would make sense as a bunch of the packages have been upgraded in this PR.

I'm not sure what you mean with points 3 & 4, I see `package-lock.json` for both sites on both `main` and `204-headless-sxa-update` branches?
* https://github.com/Sitecore/XM-Cloud-Introduction/blob/204-headless-sxa-update/src/Project/Sugcon/SugconAnzSxa/package-lock.json
* https://github.com/Sitecore/XM-Cloud-Introduction/blob/main/src/Project/Sugcon/SugconAnzSxa/package-lock.json
* https://github.com/Sitecore/XM-Cloud-Introduction/blob/204-headless-sxa-update/src/Project/Sugcon/SugconEuSxa/package-lock.json
* https://github.com/Sitecore/XM-Cloud-Introduction/blob/main/src/Project/Sugcon/SugconEuSxa/package-lock.json

Can you run a full clean of your local repo's file system after switching to the `204-headless-sxa-update` branch and try again? I usually do this by running `git clean -fxd` from the root of the repo.

sebastiantecsi commented:
@robearlam ,

I had to remove entire work folder and to clone the repo once more and run init.ps1 and up.ps1 again.

Now I see different thing:

![image](https://github.com/Sitecore/XM-Cloud-Introduction/assets/1449681/4dfd5a71-17c6-45c8-bdfb-5735183d20e8)

On my local how do I need to setup this variables:
```
SUCGON_ANZ_CDP_CLIENT_KEY=
SUCGON_ANZ_CDP_TARGET_URL=
SUCGON_ANZ_CDP_POINTOFSALE=
SUCGON_EU_CDP_CLIENT_KEY=
SUCGON_EU_CDP_TARGET_URL=
SUCGON_EU_CDP_POINTOFSALE=

```
So the my local solution works fine?

robearlam commented:
Correct, these are new variables that are required as Embedded Personalisation has been enabled in the base repo since our original cut was taken. These params have been added to the readme and init.ps1 as part of this PR, which is how you can populate them.

As for what values to use, if you have an XMC instance you can get these values from within the XMC Sites Manager, under the Dev Settings for the site you're trying to configure. If you don't have an XMC instance that you're connecting to then you can just use a set of dummy values locally, e.g. I just tested running init.ps1 with the following values and it worked.

```
-SUCGON_ANZ_CDP_CLIENT_KEY "abdc1234"
-SUCGON_ANZ_CDP_TARGET_URL "https://www.somedomain.com" 
-SUCGON_ANZ_CDP_POINTOFSALE "abdc1234" 
-SUCGON_EU_CDP_CLIENT_KEY "abdc1234" 
-SUCGON_EU_CDP_TARGET_URL "https://www.somedomain.com" 
-SUCGON_EU_CDP_POINTOFSALE "abdc1234"
```

robearlam commented:
So @sebastiantecsi the foundation head now includes the middleware/components for Embedded Personalisation. I added params to the `docker-compose.override.yml` and updated the `init.ps` and `readme` with instructions on how to populate them.

What I hadn't factored in was people who don't want to connect them locally for dev. Turns out we found a bug in the foundation head where you couldn't disable it, which I've reported internally. I've just pushed a couple more commits to this repo so that it is disabled by default as most people won't have it enabled for dev.

You'll need to pull latest and redo init to pull in the new `.env` param.

sebastiantecsi commented:
@robearlam ,

I've pulled latest and redo init to pull in the new .env param and now all is much better now, the website looks good.

Is there a reason that the folders below were added? As to me looks that are not referenced and used at all.

```
\src\Project\Sugcon\SugconEuSxa\src\assets\basic\
\src\Project\Sugcon\SugconEuSxa\src\assets\sass\
\src\Project\Sugcon\SugconAnzSxa\src\assets\basic\
\src\Project\Sugcon\SugconAnzSxa\src\assets\sass\
```
Can you please double check? Or is there a reason for having those files there?

Also, thank you for updating the repo with the latest content from production, and its clear why those items were added

```
/sitecore/media library/Project/Sugcon/SugconEuSxa/Sitemaps/sitemap
/sitecore/media library/Project/Sugcon/SugconAnzSxa/Sitemaps/sitemap
```
But I've create a new page that shows up in the website's menu, but not in the sitemap.xml page in the browser. How does it get updated?

Thanks.

robearlam commented:
Good catch @sebastiantecsi  on the `scss` files, they came in from the basic site template and aren't required so I've removed them.

With the Sitemap it's generated on publish, and as you don't perform publish locally it won't auto-generate. You can open the media library item and edit the XML content if you really need it included. If that becomes a problem, then lets track it in a separate issue so that we can get these updated merged in the meantime.

sebastiantecsi commented:
@robearlam ,

Looks good now.

Thank you for the sitemap clarification.
